Tempstr.pas

```diff
@@ -117,27 +117,34 @@ destructor TTempFileStream.Destroy;
 class function TTempFileStream.GenerateTempFileName(const ATempPath: string; const APrefix: String = ''): string;
 var
   i : integer;
+  TempFileName : PAnsiChar;
   TempPath, Prefix : string;
-  F : Text;
+  h : THandle;
 begin
-  if ATempPath[Length(ATempPath)] = '\' then
+  if (ATempPath = '') or (ATempPath[Length(ATempPath)] = '\') then
     TempPath := ATempPath
   else TempPath := ATempPath + '\';
   if APrefix = '' then
     Prefix := 'TmpFile'
   else Prefix := APrefix;
 
-  for i := 0 to 10000000 do
-    begin
-      Result := Format('%s%s_%.8d.tmp', [TempPath, APrefix, i]);
-      if not FileExists(Result) then
-        break;
-    end;
-  Assign(f, Result);
+  TempFileName := StrAlloc (Max_Path);
   try
-    ReWrite(f);
+    i := 0;
+    while True do
+      begin
+        TempFileName := PAnsiChar(Format('%s%s_%.8d.tmp', [TempPath, APrefix, i]));
```

Comment:
TempFileName should be a regular String, not a PAnsiChar. You should do the cast within the call to CreateFile